net/netcheck/netcheck.go

```diff
@@ -785,12 +792,25 @@ func (c *Client) GetReport(ctx context.Context, dm *tailcfg.DERPMap) (_ *Report,
 	c.curState = rs
 	last := c.last
 	now := c.timeNow()
+	doFull := false
 	if c.nextFull || now.Sub(c.lastFull) > 5*time.Minute {
+		doFull = true
+	}
+	// If the last report had a captive portal and reported no UDP access,
+	// it's possible that we didn't get a useful netcheck due to the
+	// captive portal blocking us. If so, make this report a full
+	// (non-incremental) one.
+	if !doFull && last != nil {
+		v, ok := last.CaptivePortal.Get()
+		doFull = v && ok && !last.UDP
```

Comment:
`doFull =  !last.UDP && last.CaptivePort.EqualBool(true)`

(cheaper stuff first and then use EqualBool instead of Get)